x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/components/flyout_components/add_to_lists_table/use_add_to_lists_table.tsx

```diff
@@ -0,0 +1,158 @@
+   * when adding an exception item from the exception lists management page, there is no
+   * list or rule to go off of, so user can select to add the exception to any rule or to any
+   * shared list.
+   */
+  showAllSharedLists: boolean;
+  /* Shared exception lists to display as options to add item to */
+  sharedExceptionLists: ListArray;
+  onListSelectionChange: (listsSelectedToAdd: ExceptionListSchema[]) => void;
+}
+
+export const useAddToSharedListTable = ({
+  showAllSharedLists,
+  sharedExceptionLists,
+  onListSelectionChange,
+}: ExceptionsAddToListsComponentProps) => {
+  const [listsToDisplay, setListsToDisplay] = useState<ExceptionListRuleReferencesSchema[]>([]);
+  const [exceptionListReferences, setExceptionListReferences] = useState<RuleReferences>({});
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const listsToFetch = useMemo(() => {
+    return showAllSharedLists ? [] : sharedExceptionLists;
+  }, [showAllSharedLists, sharedExceptionLists]);
+
+  // here we don't have initial selected lists as they component is used only in the Add Exception Flyout
+  const [linkedLists, setLinkedLists] = useState<ExceptionListRuleReferencesSchema[]>([]);
+
+  const [pagination, setPagination] = useState({
+    pageIndex: 0,
+    initialPageSize: 5,
+    showPerPageOptions: false,
+  });
+  const [message, setMessage] = useState<JSX.Element | string | undefined>(
+    <EuiLoadingContent lines={4} data-test-subj="exceptionItemListsTableLoading" />
+  );
+
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  const getReferences = useCallback(async () => {
+    try {
+      setIsLoading(true);
+      const result = await getExceptionItemsReferences(
+        (!listsToFetch.length
+          ? [{ namespace_type: 'single' }, { namespace_type: 'agnostic' }] // TODO remove 'agnostic' when using `single` only
+          : listsToFetch) as ExceptionListSchema[]
+      );
+      setExceptionListReferences(result as RuleReferences);
+    } catch (err) {
+      setError(i18n.REFERENCES_FETCH_ERROR);
+    }
+  }, [listsToFetch]);
+
+  useEffect(() => {
+    getReferences();
+  }, [listsToFetch, getReferences]);
+
+  useEffect(() => {
+    if (exceptionListReferences) {
+      const lists: ExceptionListRuleReferencesSchema[] = [];
+      for (const [_, value] of Object.entries(exceptionListReferences))
+        if (value.type === ExceptionListTypeEnum.DETECTION) lists.push(value);
+
+      setMessage(undefined);
+      setListsToDisplay(lists);
+      setIsLoading(false);
+    }
+  }, [exceptionListReferences, showAllSharedLists]);
+
+  useEffect(() => {
+    onListSelectionChange(
+      linkedLists.map(
+        ({
+          referenced_rules: _,
+          namespace_type: namespaceType,
+          os_types: osTypes,
+          tags,
+          ...rest
+        }) => ({
+          ...rest,
+          namespace_type: namespaceType ?? 'single',
```

Comment:
Why do we need this transformation?

Reply:
What I understood is that in order to send `namespacetype: single` by default, if the `list.namespace` had a  falsy value and in order to do so in the `tsx` if we used the `snack case` we will receive the below error
`Parameter name `namespace_type` must match one of the following formats: camelCase, PascalCase`